api/src/main/webapp/WEB-INF/cron_default.yaml

```diff
@@ -4,24 +4,44 @@ cron:
   schedule: every 3 hours
   timezone: UTC
   target: api
-- description: 'Daily sync of compliance training status (via Moodle API)'
+- description: |
+    Daily sync of compliance training status (via Moodle API). This is just a backup to the primary
+    flow. This would only ever catch the following scenarios:
+    1. The user's compliance training expires. The time scale on this expiration is O(years),
+    so syncing nightly is entirely acceptable.
+
+    2. The user completes compliance and doesn't return to the dashboard. We'd progress them in the
+    background. This won't make much of a difference and would have the same effect as if they had
+    navigated back to the dashboard.
+
+    3. Somehow the user manages to "uncomplete" training in Moodle. I'm not sure this is possible,
+    but if so, it's certainly an edge case where nightly would be acceptale latency
   url: /v1/cron/bulkSyncComplianceTrainingStatus
   schedule: every 24 hours
   timezone: UTC
   target: api
-- description: 'Daily sync of eRA Commons linkage status (via FireCloud API)'
+- description: >
+    Daily sync of eRA Commons linkage status (via FireCloud API).  I expect this cron to never
+    detect anything. This is effectively checking for whether any changes happened in the
+    background between the Terra account and its linkage to eRA commons. During onboarding,
+    this won't matter since it's checked live from the dashboard. After onboarding, I don't even
```

Comment:
My commentary here is way too ambiguous to be checked in. @gjuggler may be able to provide a better description. But before I wrote this I'd need to refresh on exactly what the intended behavior is. In particular, I believe the current registration system as designed, only cares that a linkage was established at one point, for verification - this is purely a stand-in for ID verification. If so, this background sync would actually behave incorrectly if a user unlinked their account, or let their linkage expire.

However, for controlled tier, there might be new policies which require an active eRA commons linkage, in case this stricter behavior would be required, including a linkage refresh.

Reply:
I don't want to go too far down a rabbit hole, but it sounds like we're collecting this information but never acting on it. It just logs things to INFO but doesn't ring any bells elsewhere as far as I can see.

Reply:
No, it's not wrong to be digging into the details here. [This line](https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/api/OfflineUserController.java#L142) should be updating user status (including registered-tier membership) based on the updated eRA Commons linkage state synced from Terra.